src/ontology.rs

```diff
@@ -745,6 +758,52 @@ impl Ontology {
         self.omim_diseases.values()
     }
 
+    /// Returns a vector contains diseases whose names contains the provided
+    /// substring.
+    ///
+    /// If no such substring is present, the vector will be empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hpo::Ontology;
+    /// let ontology = Ontology::from_binary("tests/example.hpo").unwrap();
+    ///
+    /// for result in ontology.diseases_by_name("Cystinosis") {
+    ///     println!("{:?}", result.name());
+    ///  }
+    /// ```
+    pub fn diseases_by_name<'a>(
+        &'a self,
+        symbol: &'a str,
+    ) -> DiseaseIter<impl FnMut(&&'a OmimDisease) -> bool + 'a> {
+        DiseaseIter {
+            inner: self
+                .omim_diseases
+                .values()
+                .filter(move |disease| disease.name().contains(symbol)),
+        }
+    }
+
+    /// Returns the first matching diseases whose names contains the provided
```

Comment:
Here you're mixing singular and plural forms, which makes the documentation confusing without looking at the code. I suggest to say `Returns the fist matching disease whose name contains the provided substring`